### PR TITLE
Add ability to specify custom plugins

### PR DESCRIPTION
**Pull request by iloire**
All credits to @tbasse (we paired on this). I just helped him out writing the tests and decouple `app.js` form `server.js` to increase testability.

This PR also addresses indirectly https://github.com/itteco/iframely/issues/142 :)

### CODE REVIEW

File: server.js [hunk 1]
```diff
@@ -1,158 +1,7 @@
...
-    res.writeHead(302, { Location: 'http://iframely.com'});
-    res.end();
-});
-
-process.title = "iframely";
-process.env.NODE_TLS_REJECT_UNAUTHORIZED = "0";
-
-var listener = app.listen(process.env.PORT || CONFIG.port, process.env.HOST || CONFIG.host, function(){
-    console.log('\niframely is listening on ' + listener.address().address + ':' + listener.address().port + '\n');
+var server = app.listen(process.env.PORT || CONFIG.port, process.env.HOST || CONFIG.host, function(){
```
Comment (iloire) [inline server.js]: Now ``server.js`` only loads the express app, so the app lifecycle is much easier to control

File: test/e2e.js [hunk 2]
```diff
@@ -8,19 +8,24 @@ var async = require('async');
 describe('meta endpoint', function() {
 
   var BASE_IFRAMELY_SERVER_URL = 'http://localhost:' + process.env.PORT;
-  var server = require('../server'); // start card meta
```
Comment (iloire) [inline test/e2e.js]: These tests got a bit better. Since we control the app lifecycle we can start and shutdown the server for each test.

**Comment by iparamonau** [comment]
Thanks guys. Apologies for the delay - we've got distracted with couple requests over here. We'll merge early next week.

File: app.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
+var sysUtils = require('./utils');
+
+console.log("");
```
Comment (iloire) [inline app.js]: ``app.js`` has just been extracted our from ``server.js``, without making any other changes into its internal workings.

**Comment by iloire** [comment]
Thanks @iparamonau!

**Pull request merged**